pkg/acs/calacs/acsccd/doccd.c

```diff
@@ -251,109 +257,291 @@ int DoCCD (ACSInfo *acs_info) {
+                  }
+
+                  driftcorr = NO;
+
+                  /* Unsupported subarray apertures and other oddities.  These
+                     are processed using the original/old bias correction algorithm.
+                  */ 
+                  } else {
+                     trlmessage("WFC readout type/gain not set as needed for new bias level algorithm for subarray data.");
+                     trlmessage("Data to be processed with original bias level algorithm.");
+
+                     for (i = 0; i < acs_info->nimsets; i++) {
+
+                         /* This is set based on results of FindOver */
+                         done = overscan[i];
+
+                         if (doBlev(&acs[i], &x[i], acs[i].chip, &meanblev, &done,
+                                 &driftcorr))
+                            return (status);
+
+                         if (done) {
+                            trlmessage("Bias level from overscan has been subtracted;");
+
+                            if (PutKeyFlt(&x[i].sci.hdr, "MEANBLEV", meanblev,
+                                           "mean of bias levels subtracted in electrons")) {
+                               return (status);
+                            }
+
+                            sprintf(MsgText, "     mean of bias levels subtracted was %.6g electrons.", meanblev);
+                            trlmessage(MsgText);
+
+                         } else {
+                          trlmessage("Default bias level from CCDTAB was subtracted.");
+                         }
+
+                         /* Provide immediate feedback to the user on the values computed
+                            for each AMP, but only for those amps used for the chip being
+                            processed.
+                         */
+                         for (j = 0; j < NAMPS; j++) {
+                             if (acs[i].blev[j] != 0.) {
+                                 sprintf(MsgText, "     bias level of %.6g electrons was subtracted for AMP %c.", acs[i].blev[j], AMPSORDER[j]);
+                                 trlmessage(MsgText);
+
+                                 acs_info->blev[j] = acs[i].blev[j];
+                             }
+                         }
+
+                         /* Set this to complete so overscan-trimmed image will be written out. */
+                         blevcorr[i] = COMPLETE;
+                     }
+  
+                     PrSwitch("blevcorr", COMPLETE);
+  
+                     if (acs_info->printtime)
+                        TimeStamp("BLEVCORR complete", acs_info->rootname);
+
+                     driftcorr = NO;
+                  } /* End subarray processing */
+
+            } /* End of full frame/subarray data */
+
+          } /* End of Pre-/Post-SM4 data */
+
+       } /* End of only WFC detector */
+
+       /* All HRC and SBC data - use the original blev correction */
```

Comment:
SBC data don't get processed here. This is only for CCD, not MAMA. (Please correct me if I misremember.)

Reply:
Ooops.  I got a bit too inclusive in my comments - here there be no SBC.